client/src/components/UserTrainingPlans.jsx

```diff
@@ -1,24 +1,79 @@
 //Import necessary functionalities
-import React, { useContext, useState } from 'react';
-import ComponentContext from '../ComponentContext.js';
+import React, { useState, useEffect } from 'react';
+import { UserStatus } from '../context/UserContext';
+import { useNavigate } from 'react-router-dom';
 
-const UserTrainingPlans = () => {
+const UserTrainingPlans = ({ selectedPet }) => {
+    
+    //Use useNavigate hook from React Router to establish functionality in buttons below
+    const navigate = useNavigate();
+    
+    //Import user settings to use in data pulls
+    const { loggedUser, setLoggedUser } = UserStatus();
     
-    const { setComponent } = useContext(ComponentContext);
-
-    //Will need to pull all training plan information for this particular user via server GET request
     //Hold training plan info in state
-    const [trainingPlans, setTrainingPlans] = useState([]);
+    const [trainingPlans, setTrainingPlans] = useState(null);
+
+    //Function to pull all training plans for selected pet
+    const pullPlans = async () => {
+        //Reset previous training plans from other pets first
+        setTrainingPlans(null);
+        
+        //Pull and store query parameters
+        const userId = loggedUser.userId;
+        const petId = selectedPet.petId; //selectedPet is an object that is {petId: petId, petName: petName}
+
+        //Send query with user and pet parameters
+        const response = await fetch(`api/plans/${userId}/${petId}`);
+
+        //Save response in a variable to access for display
+        const trainingPlansDetails = await response.json();
+
+        if (!trainingPlansDetails[0].subscriptions) {
+            setTrainingPlans(null);
+        } else {
+            //Remove the first element of the array (subscription status)
+            trainingPlansDetails.shift();
```

Comment:
There is a dependency that the subscription state will always be here. Same with assuming element 0 above. I feel like there should be a cleaner way of doing this instead of the assumption. You could return an object where it passes a subscription property and a trainingPlanDetails property.

Reply:
I think I understand what you mean but may need to chat about it more at our 1:1 next week during code clean up. I added this to my notes to make sure I remember to look back at this comment. 

For us to review: 
Right now, the server returns an array structured as `[ { subscriptions: true }, ...allSubscriptionInfo ]`. I tried to change it to make it return an object as `{ subscriptions: true, trainingPlansDetails: {} }` and though it works server side (tested through Postman, example below), I am struggling to implement it client-side because my current logic iterates through the array of ..allSubscriptionInfo. Going from object -> array is not coming to me right now, so I added a note for myself to tackle next week!

Postman Screenshot of Potential Future Change:
<img width="991" alt="Screenshot 2024-10-30 at 2 59 42 PM" src="https://github.com/user-attachments/assets/e4665acd-f9e8-4c62-b0c2-93262e119b34">